include/openthread/ble_secure.h

```diff
@@ -0,0 +1,469 @@
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the top-level functions for the OpenThread BLE Secure implementation.
+ *
+ *  @note
+ *   The functions in this module require the build-time feature `OPENTHREAD_CONFIG_BLE_TCAT_ENABLE=1`.
+ *
+ *  @note
+ *   To enable cipher suite DTLS_PSK_WITH_AES_128_CCM_8, MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
+ *    must be enabled in mbedtls-config.h
+ *   To enable cipher suite DTLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,
+ *    MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED must be enabled in mbedtls-config.h.
+ */
+
+#ifndef OPENTHREAD_BLE_SECURE_H_
+#define OPENTHREAD_BLE_SECURE_H_
+
+#include <stdint.h>
+#include <openthread/message.h>
+#include <openthread/tcat.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup api-ble-secure
+ *
+ * @brief
+ *   This module includes functions that control BLE Secure (TLS over BLE) communication.
+ *
+ *   The functions in this module are available when BLE Secure API feature
+ *   (`OPENTHREAD_CONFIG_BLE_TCAT_ENABLE`) is enabled.
+ *
+ * @{
+ *
+ */
+
+/**
+ * This function pointer is called when the secure BLE connection state changes.
+ *
+ * @param[in]  aInstance            A pointer to an OpenThread instance.
+ * @param[in]  aConnected           TRUE, if a secure connection was established, FALSE otherwise.
+ * @param[in]  aBleConnectionOpen   TRUE if a BLE connection was established to carry a TLS data stream, FALSE
+ *                                  otherwise.
+ * @param[in]  aContext             A pointer to arbitrary context information.
+ *
+ */
+typedef void (*otHandleBleSecureConnect)(otInstance *aInstance,
+                                         bool        aConnected,
+                                         bool        aBleConnectionOpen,
+                                         void       *aContext);
+
+/**
+ * This function pointer is called when data was received over a BLE Secure TLS connection.
+ * When TLV mode is activate, the function will be called once a complete TLV was received and the
+ * message offset points to the TLV value.
+ *
+ *
+ * @param[in]  aMessage          A pointer to the message.
+ * @param[in]  aTcatMessageType  The message type received.
+ * @param[in]  aServiceName      The name of the service the message is direced to.
+ * @param[in]  aContext          A pointer to arbitrary context information.
+ *
+ */
+typedef otHandleTcatApplicationDataReceive otHandleBleSecureReceive;
+
+/**
+ * This function starts the BLE Secure service.
+ * When TLV mode is activate, the function will be called once a complete TLV was received and the
+ * message offset points to the TLV value.
```

Comment:
Agreed. False copy from above.

Reply:
I wanted to double-check if the comment now shows up above the right function! It could also be useful to have this info above the `typedef otHandleTcatApplicationDataReceive otHandleBleSecureReceive;` line ? That would explain when the function is called. 

Also the text "the function will be called" can be confusing because there are multiple functions this could refer to. Maybe say "When TLV mode is **active**, the function **aReceiveHandler** will be called once ... ".